lib/tasks/tasks.rake

```diff
@@ -9,4 +9,9 @@ namespace :tasks do
       end
     end
   end
+
+  desc "Sync subjects"
+  task sync_subjects: :environment do
+    Notification.subjectable.without_subject.each{|n| n.send(:update_subject) }
```

Comment:
Good catch, used `find_each` instead.

The 2 second bit isn't for timing or delaying downloads, it's only used when updating existing downloaded subjects to check if the newly synced notification was created around (within 2 seconds) the same time as the time was last saw that the subject was updated, if so we know there's no need to fetch an update.

I'd suspect that the `sync_subjects` rake task would tend to hit peoples rate limits the first time around, the nice thing about using the `without_subject` scope is that it can be ran multiple times and will only download stuff it didn't get the first time around.